.github/workflows/build-docker.yml

```
@@ -0,0 +1,143 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*.*.*'
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to build and deploy'
+        required: false
+
+env:
+  REPO: 'artifact.aerospike.io/ecosystem-container-dev-local'
+  JFROG_CLI_BUILD_NAME: 'asvec'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Install JFrog CLI
+        uses: jfrog/setup-jfrog-cli@v3
+        with:
+          version: 'latest'
+
+      - name: Login to Artifact Aerospike Docker Registry
+        run: |
+            jf docker login artifact.aerospike.io --username ${{ secrets.JFROG_USERNAME }} --password ${{ secrets.JFROG_ACCESS_TOKEN }}
+            docker pull artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
+            docker push artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
```

```suggestion
            # needed these to get the docker cache working
            docker pull artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
            docker push artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
```
I think a comment about why this is needed would be good.